app/lib/shared/task_scheduler.dart

```diff
@@ -49,37 +44,16 @@ class ManualTriggerTaskSource extends TaskSource {
     });
   }
 
+  @override
+  Stream<Task> startStreaming() => _controller.stream;
```

Comment:
To simplify the code in the constructor, you can actually just listen on the port here:
```dart
Stream<Task> startStreaming() async* {
  await for (final Task task in _taskReceivePort) {
    // Could sleep here to throttle if wanted.
    yield task;
  }
}
```